code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -1121,142 +1143,140 @@ var/global/list/damage_icon_parts = list()
+				under.Insert(new/icon(tail_s, dir=SOUTH), dir=SOUTH)
+				under.Insert(new/icon(tail_s, dir=EAST), dir=EAST)
+				under.Insert(new/icon(tail_s, dir=WEST), dir=WEST)
 
-				overlays_standing[TAIL_UNDERLIMBS_LAYER]	= image(temp1)
+				overlays_standing[TAIL_UNDERLIMBS_LAYER] = image(under)
 
-				// Creates a blank icon, and copies accessory_s' north direction sprite into it
-				// before passing that to the tail layer that overlays uniforms and such.
-				var/icon/temp2 = new/icon("icon" = 'icons/effects/species.dmi', "icon_state" = "[species.tail]_delay")
-				temp2.Insert(new/icon(tail_s,dir=NORTH),dir=NORTH)
+				// Creates a blank icon, and copies accessory_s' north direction sprite into it before passing that to the tail layer that overlays uniforms and such.
+				var/icon/over = new/icon("icon" = 'icons/effects/species.dmi', "icon_state" = "blank")
+				over.Insert(new/icon(tail_s, dir=NORTH), dir=NORTH)
 
-				overlays_standing[TAIL_LAYER]	= image(temp2)
+				overlays_standing[TAIL_LAYER] = image(over)
 			else // Otherwise, since the user's tail isn't overlapped by limbs, go ahead and use default icon generation.
-				overlays_standing[TAIL_LAYER]	= image(tail_s)
+				overlays_standing[TAIL_LAYER] = image(tail_s)
 
 	if(update_icons)
 		update_icons()
 
 
 /mob/living/carbon/human/proc/start_tail_wagging(var/update_icons=1)
 	overlays_standing[TAIL_UNDERLIMBS_LAYER] = null // SEW direction icons, overlayed by LIMBS_LAYER.
-	overlays_standing[TAIL_LAYER] = null // This will be one of two things:
-										 // If the species' tail is overlapped by limbs, this will be only the N direction icon so tails can still appear on the outside of uniforms and such.
-										 // Otherwise, since the user's tail isn't overlapped by limbs, it will be a full icon with all directions.
+	overlays_standing[TAIL_LAYER] = null /* This will be one of two things:
+											If the species' tail is overlapped by limbs, this will be only the N direction icon so tails can still appear on the outside of uniforms and such.
+											Otherwise, since the user's tail isn't overlapped by limbs, it will be a full icon with all directions. */
+
+	var/list/marking_styles = params2list(m_styles)
+	var/icon/tail_marking_icon
+	var/datum/sprite_accessory/body_markings/tail/tail_marking_style
+	if(marking_styles["tail"] != "None" && (species.bodyflags & HAS_TAIL_MARKINGS))
+		var/tail_marking = marking_styles["tail"]
+		var/list/marking_colours = params2list(m_colours)
+		marking_colours["tail"] = sanitize_hexcolor(marking_colours["tail"])
+		tail_marking_style = marking_styles_list[tail_marking]
+		tail_marking_icon = new/icon("icon" = tail_marking_style.icon, "icon_state" = "[tail_marking_style.icon_state]w_s")
+		tail_marking_icon.Blend(rgb(hex2num(copytext(marking_colours["tail"], 2, 4)), hex2num(copytext(marking_colours["tail"], 4, 6)), hex2num(copytext(marking_colours["tail"], 6, 8))), ICON_ADD)

```

Comment:
hexcolor->R/G/B->hexcolor

I'm going to stop leaving comments on these, because this is just getting silly. You should be able to just search for `rgb(hex2num` and find every instance of this.